aiohttp_client_cache/backends/sqlite.py

```diff
@@ -57,45 +55,36 @@ def __init__(
     ):
         super().__init__(**kwargs)
         self.connection_kwargs = get_valid_kwargs(sqlite_template, kwargs)
+        self.connection_kwargs['check_same_thread'] = False
         self.filename = _get_cache_filename(filename, use_temp)
         self.table_name = table_name
 
         self._bulk_commit = False
-        self._initialized = False
-        self._connection = None
+        self._connection: Optional[aiosqlite.Connection] = None
         self._lock = RLock()
 
     @asynccontextmanager
-    async def get_connection(self, autocommit: bool = False) -> AsyncIterator[aiosqlite.Connection]:
-        db = (
-            self._connection
-            if self._connection
-            else await aiosqlite.connect(self.filename, **self.connection_kwargs)
-        )
-        try:
-            yield await self._init_db(db)
-            if autocommit and not self._bulk_commit:
-                await db.commit()
-        finally:
-            if not self._bulk_commit:
-                await db.close()
+    async def get_connection(self, commit: bool = False) -> AsyncIterator[aiosqlite.Connection]:
+        if not self._connection:
+            self._connection = await aiosqlite.connect(self.filename, **self.connection_kwargs)
+            await self._init_db()
+        yield self._connection
+        if commit and not self._bulk_commit:
+            await self._connection.commit()
 
-    async def _init_db(self, db: aiosqlite.Connection):
-        """Create table if this is the first connection opened, and set fast save if possible"""
+    async def _init_db(self):
+        """Initialize the database, if it hasn't already been"""
         with self._lock:
```

Comment:
This is gonna block the event loop, right?

Reply:
Right, which is why I'm considering dropping thread-safety for this backend. There are valid use cases for async + multithreading, but since the other backends are better suited for that, I think going back to using only `asyncio.Lock` may be the best option. I'll just need to add some notes about that to the docs.

What do you think?

Reply:
I have reverted to `asyncio.Lock`.